LDKNodeMonday/Service/KeyService/KeyService.swift

```diff
@@ -41,28 +41,26 @@ private struct KeyService {
 }
 
 extension KeyService {
-    func saveNetwork(network: String) throws {
-        keychain[string: "SelectedNetwork"] = network
+    func saveNetwork(networkString: String) throws {
+        let currentBackupInfo = try self.getBackupInfo()
+        let newBackupInfo = BackupInfo(mnemonic: currentBackupInfo.mnemonic, networkString: networkString, serverURL: currentBackupInfo.serverURL)
+        try self.saveBackupInfo(backupInfo: newBackupInfo)
     }
 
     func getNetwork() throws -> String? {
-        return keychain[string: "SelectedNetwork"]
+        let backupInfo = try self.getBackupInfo()
+        return backupInfo.networkString
     }
 
-    func deleteNetwork() throws {
-        try keychain.remove("SelectedNetwork")
+    func saveServerURL(url: String) throws {
+        let currentBackupInfo = try self.getBackupInfo()
+        let newBackupInfo = BackupInfo(mnemonic: currentBackupInfo.mnemonic, networkString: currentBackupInfo.networkString, serverURL: url)
+        try self.saveBackupInfo(backupInfo: newBackupInfo)
```

Comment:
for `saveNetwork` and `saveServerURL` does it make sense for us to just have one `saveBackupInfo` method instead now since all of those methods under the hood are saving the whole `BackupInfo` (and not just specific parts like network alone anymore)? 

Not a blocker to this PR, but something I'd at least like your thoughts on to update in this PR or follow up with a PR about if it avoids the redundant loads/saves (even though the tradeoff is very slightly more verbose call sites) potentially. Thoughts?

Reply:
saveServer and saveNetwork are handy for the situations where we are (and will be, in the network switch PR) have UI for selecting the server and network independent of the mnemonic. There are moments where only one of them is changed, or there some time before setting both.

It saves bringing the mnemonic out in code places where it is only retrieved to be used to save one of the other values. I felt it made sense to keep the saveNetwork() and saveServer() functions that existed rather than complicate things further.

But if you have specific suggestions for optimisation that would be good.